src/backends/backend/tensorflow/tensorflow.cc

```diff
@@ -0,0 +1,1923 @@
+      (std::string("TRITONBACKEND_ModelExecute: model ") + name_ +
+       " released " + std::to_string(request_count) + " requests")
+          .c_str());
+}
+
+}  // namespace
+
+/////////////
+
+extern "C" {
+
+// Implementing TRITONBACKEND_ModelInitialize is optional. The backend
+// should initialize any state that is intended to be shared across
+// all instances of the model.
+TRITONSERVER_Error*
+TRITONBACKEND_ModelInitialize(TRITONBACKEND_Model* model)
+{
+  const char* cname;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelName(model, &cname));
+  std::string name(cname);
+
+  uint64_t version;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelVersion(model, &version));
+
+  TRITONSERVER_LogMessage(
+      TRITONSERVER_LOG_INFO, __FILE__, __LINE__,
+      (std::string("TRITONBACKEND_ModelInitialize: ") + name + " (version " +
+       std::to_string(version) + ")")
+          .c_str());
+
+  // With each model we create a ModelState object and associate it
+  // with the TRITONBACKEND_Model.
+  ModelState* model_state;
+  RETURN_IF_ERROR(ModelState::Create(model, &model_state));
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelSetState(model, reinterpret_cast<void*>(model_state)));
+
+  return nullptr;  // success
+}
+
+// Implementing TRITONBACKEND_ModelFinalize is optional unless state
+// is set using TRITONBACKEND_ModelSetState. The backend must free
+// this state and perform any other cleanup.
+TRITONSERVER_Error*
+TRITONBACKEND_ModelFinalize(TRITONBACKEND_Model* model)
+{
+  void* vstate;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelState(model, &vstate));
+  ModelState* model_state = reinterpret_cast<ModelState*>(vstate);
+
+  TRITONSERVER_LogMessage(
+      TRITONSERVER_LOG_INFO, __FILE__, __LINE__,
+      "TRITONBACKEND_ModelFinalize: delete model state");
+
+  delete model_state;
+
+  return nullptr;  // success
+}
+
+// Implementing TRITONBACKEND_ModelExecute is required.
+TRITONSERVER_Error*
+TRITONBACKEND_ModelExecute(
+    TRITONBACKEND_Model* model, TRITONBACKEND_Request** requests,
+    const uint32_t request_count)
+{
+  const char* model_name;
+  RETURN_IF_ERROR(TRITONBACKEND_ModelName(model, &model_name));
+
+  TRITONSERVER_LogMessage(
+      TRITONSERVER_LOG_VERBOSE, __FILE__, __LINE__,
+      (std::string("TRITONBACKEND_ModelExecute: model ") + model_name +
+       " with " + std::to_string(request_count) + " requests")
+          .c_str());
+
+  // Triton only calls model execute from a single thread at a time
+  // *for a given model*. But since this backend could be used by
+  // multiple models the implementation needs to handle multiple
+  // models executing at the same time. Good practice for this is to
+  // use only function-local and model-specific state (obtained from
+  // 'model'), which is what we do here.
+  ModelState* state;
+  RETURN_IF_ERROR(
+      TRITONBACKEND_ModelState(model, reinterpret_cast<void**>(&state)));
+
+  // At this point we accept ownership of 'requests', which means that
```

Comment:
There is another "ownership accepting" note above... I think this one is correct but just need to make sure ownership is handled correctly.

Reply:
Are you referring to the comment [here](https://github.com/NVIDIA/triton-inference-server/blob/9d30c5a7be030f53cb83fafd9d980fa5ddf60196/src/backends/backend/tensorflow/tensorflow.cc#L1431-L1441)? I think both are correct, the other note is actually saying that the request will not be released until the end even if error occurs during the execution.

Reply:
Yes, you are right.